src/main/java/seedu/address/logic/parser/EditApptCommandParser.java

```diff
@@ -76,6 +76,22 @@ public EditApptCommand parse(String args) throws ParseException {
         if (argMultimap.getValue(PREFIX_NEW_END_TIME).isPresent()) {
             editApptDescriptor.setEndTime(ParserUtil.parseTime(argMultimap.getValue(PREFIX_NEW_END_TIME).get()));
         }
+
+        //Check if valid new time period (both new start and end time given)
+        if (argMultimap.getValue(PREFIX_NEW_END_TIME).isPresent()
+                && argMultimap.getValue(PREFIX_NEW_START_TIME).isPresent()) {
+            ParserUtil.parseTimePeriod(argMultimap.getValue(PREFIX_NEW_START_TIME).get(),
+                    argMultimap.getValue(PREFIX_NEW_END_TIME).get());
+        }
+
+        //extra check to parse if new end time before current start time (only new end time given)
+        //will check the vice versa in command itself as curr end time not an input
+        if (argMultimap.getValue(PREFIX_NEW_END_TIME).isPresent()
```

Comment:
Since EditApptCommand throws CommandException when the TimePeriod is invalid, not sure if the additional checks added from line 80 to 93 here are redundant?

Reply:
True! Was thinking of it as a way to check before it gets passed to the EditApptCommand, not sure if it would help or reduce performance. Should I perhaps remove the extra check then?

Reply:
I see, that works too! But I guess the extra check wouldn't account for the case where only the new start time is provided and it exceeds the appointment's current end time (not available to parser). In that case, not sure if the extra checks might not help much and should be removed instead. What do you think?

Reply:
Sounds good, will remove it 👍